src/pageEditor/slices/editorSlice.ts

```diff
@@ -446,6 +446,15 @@ export const editorSlice = createSlice({
       const uuid = action.payload;
       removeElement(state, uuid);
     },
+    removeAllElementsForRecipe(state, action: PayloadAction<RegistryId>) {
+      const recipeId = action.payload;
+      const elementIdsToRemove = state.elements
+        .filter((element) => element.recipe?.id === recipeId)
+        .map(({ uuid }) => uuid);
+      for (const uuid of elementIdsToRemove) {
+        removeElement(state, uuid);
+      }
```

Comment:
This can be one loop:
```suggestion
      for (const uuid of state.elements) {
        if (element.recipe?.id === recipeId) {
          removeElement(state, element.uuid);
        }
      }
```